java/ddlogapi.c

```diff
@@ -268,37 +362,41 @@ JNIEXPORT void JNICALL Java_ddlogapi_DDlogAPI_ddlog_1flatbuf_1free(
             ((size_t)(*env)->GetDirectBufferCapacity(env, buf)) + offset);
 }
 
-
-JNIEXPORT jint JNICALL Java_ddlogapi_DDlogAPI_ddlog_1transaction_1rollback(
+JNIEXPORT void JNICALL Java_ddlogapi_DDlogAPI_ddlog_1transaction_1rollback(
     JNIEnv * env, jobject obj, jlong handle) {
-    return ddlog_transaction_rollback((ddlog_prog)handle);
+    if (ddlog_transaction_rollback((ddlog_prog)handle) < 0) {
+        throwDDlogException(env);
+    }
 }
 
-JNIEXPORT jint JNICALL Java_ddlogapi_DDlogAPI_ddlog_1apply_1updates(
+JNIEXPORT void JNICALL Java_ddlogapi_DDlogAPI_ddlog_1apply_1updates(
     JNIEnv *env, jclass obj, jlong progHandle, jlongArray commandHandles) {
     jlong *a = (*env)->GetLongArrayElements(env, commandHandles, NULL);
     size_t size = (*env)->GetArrayLength(env, commandHandles);
     ddlog_cmd** updates = malloc(sizeof(ddlog_cmd*) * size);
     for (size_t i = 0; i < size; i++)
         updates[i] = (ddlog_cmd*)a[i];
-    int result = ddlog_apply_updates(
-        (ddlog_prog)progHandle, updates, size);
+    if (ddlog_apply_updates((ddlog_prog)progHandle, updates, size) < 0) {
```

Comment:
check if updates is not null too. I don't know if you also want to check the free.